### PR TITLE
Export DataTree to csv through showSaveDialog

### LINKED ISSUE #66: Export to CSV not working (DATA_TREE views)
Export to CSV is not working in VsCode for views of type DATA_TREE, which are, for example, be used for Latency views when using Trace Compass server. The gif shows how to trigger it, but there is no way to know if it was successful, what the filename is and where it's stored (if actually was executed)

vscode-export-cvs-bug

### PR DESCRIPTION
**Pull request by hriday-panchasara**
Webviews run in isolated contexts that cannot access local resources. Because of this webviews could an issue with opening hyperlinks.
- Suggested fix is to use showSaveDialog from vscode api
- Users can choose file name/path
- Show success/error message after
- Depends on signal added in https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/938

Fixes #66

### CODE REVIEW

**Comment by marco-miller** [comment]
@hriday-panchasara, should this PR be set to Draft for you to rework, given its failing CI build herein?

**Comment by hriday-panchasara** [comment]
> @hriday-panchasara, should this PR be set to Draft for you to rework, given its failing CI build herein?

Yup, keeping it as draft until its dependency (https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/938) gets reviewed

File: vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx [hunk 1]
```diff
@@ -100,11 +105,13 @@ class TraceViewerContainer extends React.Component<{}, VscodeAppState>  {
   componentDidMount(): void {
       this._signalHandler.notifyReady();
       signalManager().on(Signals.ITEM_PROPERTIES_UPDATED, this._onProperties);
+      signalManager().on(Signals.SAVE_DATATREE_CSV, this._onSaveDatatreeCSV);
```
[the remark this answers is not in the document]
Comment (alexdoan3011) [reply vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx:108]: I'm submitting my pull request that adds component-based drop down menu and export table for xy-output-component. Should I change the string of the signal from 'save datatree as csv' to 'save as csv' in my PR as well?

**Review by bhufmann — CHANGES_REQUESTED** [review]

File: vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhufmann) [reply vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx]: This is resolved with the latest update of the PR.

**Review by bhufmann — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**